packages/zipkin/test/batch-recorder.test.js

```diff
@@ -302,8 +302,35 @@ describe('Batch Recorder - integration test', () => {
         'http.status_code': '200'
       }
     });
+  });
+
+  it('should only flush spans when calling flush method', () => {
+    const clock = lolex.install();
+    recorder = new BatchRecorder({
+      logger: {
+        logSpan: (span) => {
+          spans.push(JSON.parse(JSON_V2.encode(span)));
+        }
+      }
+    });
+
+    recorder.record(record(rootId, now(), new Annotation.ServerRecv()));
 
     expect(spans).to.be.empty; // eslint-disable-line no-unused-expressions
+
+    clock.tick(100); // 1000 is de batching interval
+
+    expect(spans).to.be.empty; // eslint-disable-line no-unused-expressions
+
+    recorder.flush();
+
+    expect(popSpan()).to.deep.equal({
+      traceId: rootId.traceId,
+      id: rootId.spanId,
+      kind: 'SERVER'
```

Comment:
surprised the timestamp didn't make it!

Reply:
I guess it was because of `lolex`. I think this is ready to be merged.